internal/dashboard/business/metric/metric.go

```diff
@@ -104,13 +105,35 @@ func extractMetricData(name string, resp *external.PrometheusQueryRangeResponse)
 			v, err := strconv.ParseFloat(value[1].(string), 64)
 			if err != nil {
 				logger.Warnf("failed to parse value %v", value)
+				v = 0
 			} else if math.IsNaN(v) {
-				logger.Debugf("skip NaN value at timestamp %f", t)
-			} else {
-				values = append(values, response.MetricValue{
-					Timestamp: t,
-					Value:     v,
-				})
+				logger.Debugf("value at timestamp %f is NaN, set to 0", t)
+				v = 0
+			}
+			values = append(values, response.MetricValue{
+				Timestamp: t,
+				Value:     v,
+			})
+		}
+		lenValues := len(values)
+		if lenValues == 0 {
+			continue
+		}
+		for i := range values {
+			// interpolate zero slot with average of previous and next value
+			if values[i].Value == 0 {
+				switch i {
+				case 0:
```

Comment:
is it possible to solve the problem in front end, it's really normal to lose some datapoint

Reply:
No, I asked Yangon. The component he used in frontend can not handle the NaN. Interpolation is an inevitable choice.